### DISCUSSION TITLE
[@types/jquery] event listeners deprecated since 3.3?

### DISCUSSION THREAD

**Discussion by Sixish** [DefinitelyTyped/DefinitelyTyped#57311] [Issues with a @types package] [answered]
[@types/jquery] JSDocs claim that jQuery's events (e.g. click, focusout, etc.) methods have been deprecated since 3.3 in favor of on() and trigger().

Is this a decision of DT or jQuery? The jQuery documentation doesn't say anything about these being deprecated.

[a comment by an automated account is not shown]

**Comment by leonard-thieu** [comment, marked as answer]
> [@types/jquery] JSDocs claim that jQuery's events (e.g. click, focusout, etc.) methods have been deprecated since 3.3 in favor of on() and trigger().
> 
> Is this a decision of DT or jQuery? The jQuery documentation doesn't say anything about these being deprecated.

[jQuery 3.3.0 – A fragrant bouquet of deprecations and…is that a new feature?](https://blog.jquery.com/2018/01/19/jquery-3-3-0-a-fragrant-bouquet-of-deprecations-and-is-that-a-new-feature/#:~:text=Event%20aliases%20(%233214%2C%20022b69a4))

Keep in mind that in this case, the event methods are not being removed, only discouraged from use.

**Comment by mgol** [comment]
It took us a while but, finally, event shorthand methods are marked as deprecated on the API site. See e.g.: https://api.jquery.com/click-shorthand/
